plan/task/cacheconfig.go

```diff
@@ -0,0 +1,234 @@
+	buildkit "github.com/moby/buildkit/client"
+	"github.com/moby/buildkit/cmd/buildctl/build"
+	"github.com/rs/zerolog/log"
+	"go.dagger.io/dagger/compiler"
+	"go.dagger.io/dagger/plancontext"
+	"go.dagger.io/dagger/solver"
+)
+
+func init() {
+	Register("CacheConfig", func() Task { return &cacheConfigTask{} })
+}
+
+type cacheConfigTask struct {
+}
+
+func (t cacheConfigTask) Run(ctx context.Context, pctx *plancontext.Context, _ *solver.Solver, v *compiler.Value) (*compiler.Value, error) {
+	lg := log.Ctx(ctx)
+
+	// TODO: this could probably be cleaner by making a struct and using .Decode
+
+	// imports
+	if imports := v.Lookup("imports"); imports.Exists() {
+		importFields, err := v.Lookup("imports").Fields()
+		if err != nil {
+			return nil, fmt.Errorf("failed to lookup cache imports: %w", err)
+		}
+		var importConfigStrs []string
+		for _, importField := range importFields {
+			importField := importField.Value
+			typ, err := importField.Lookup("type").String()
+			if err != nil {
+				return nil, fmt.Errorf("failed to lookup cache import type: %w", err)
+			}
+			configStr := "type=" + typ
+			switch typ {
+			case "gha":
+				if scopeField := importField.Lookup("scope"); scopeField.Exists() {
+					var err error
+					scope, err := scopeField.String()
+					if err != nil {
+						return nil, fmt.Errorf("failed to lookup GHA cache import scope: %w", err)
+					}
+					configStr += ",scope=" + scope
+				}
+
+				if urlVal := v.Lookup("url"); urlVal.Exists() {
+					var err error
+					url, err := urlVal.String()
+					if err != nil {
+						return nil, fmt.Errorf("failed to lookup GHA cache import url: %w", err)
+					}
+					configStr += ",url=" + url
+				}
+
+				// TODO: this should be a secret, not string
+				if tokenVal := v.Lookup("token"); tokenVal.Exists() {
+					var err error
+					token, err := tokenVal.String()
+					if err != nil {
+						return nil, fmt.Errorf("failed to lookup GHA cache import token: %w", err)
+					}
+					configStr += ",token=" + token
+				}
+
+			case "registry":
+				ref, err := v.Lookup("ref").String()
+				if err != nil {
+					return nil, fmt.Errorf("failed to lookup registry cache import ref: %w", err)
+				}
+				configStr += ",ref=" + ref
+			default:
+				return nil, fmt.Errorf("unsupported cache import type: %s", typ)
+			}
+			importConfigStrs = append(importConfigStrs, configStr)
+		}
+
+		// TODO: remove or lower to debug
+		lg.Info().Msgf("import cache config: %+v", importConfigStrs)
+
+		importOpts, err := parseImportCache(importConfigStrs)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse cache import config: %w", err)
+		}
+		pctx.CacheConfig.SetImports(importOpts)
+	}
+
+	// exports
+	if exportField := v.Lookup("export"); exportField.Exists() {
+		typ, err := exportField.Lookup("type").String()
+		if err != nil {
+			return nil, fmt.Errorf("failed to lookup cache export type: %w", err)
+		}
+		exportConfigStr := "type=" + typ
+		switch typ {
```

Comment:
Is it a typo or just to avoid conflict with `type`? You could name it `exportType` too?